### PR TITLE
Gui: Added FreeTurntable orbit style

### PR DESCRIPTION
**Pull request by nab-os**
Adding a second Turntable orbit style but without primary rotation selection.
This aims to offer an environment similar to Blender.

Thank you for creating a pull request to contribute to FreeCAD! Place an "X" in between the brackets below to "check off" to confirm that you have satisfied the requirement, or ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10) if there is something you don't understand.

- [X]  Your Pull Request meets the requirements outlined in section 5 of [CONTRIBUTING.md](https://github.com/FreeCAD/FreeCAD/blob/master/CONTRIBUTING.md) for a Valid PR

Please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [1.0 Changelog Forum Thread](https://forum.freecad.org/viewtopic.php?f=10&t=69438).

---

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8048">pipeline status</a> for feature branch [PR_8048](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8048). Pipeline [723462308](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/723462308) was triggered at [664aca7](https://github.com/FreeCAD/FreeCAD/pull/8048/commits/664aca726f58cedfeea55a2782bf1212ed94db22). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by luzpaz** [comment]
I'm pretty sure the `.ts file gets overwrittem when it's generated from Qt's lupdate command.

File: src/Gui/Language/FreeCAD.ts [hunk 1]
```diff
@@ -4758,8 +4758,9 @@ Select a set and then press the button to view said configurations.</source>
     <message>
         <location filename="../DlgSettingsNavigation.ui" line="280"/>
         <source>Rotation orbit style.
-Trackball: moving the mouse horizontally will rotate the part around the y-axis
-Turntable: the part will be rotated around the z-axis.</source>
+Trackball: moving the mouse horizontally will rotate the part around the y-axis.
```
Comment (luzpaz) [inline src/Gui/Language/FreeCAD.ts:4761]: Please revert the changes in the .ts file (it's autogenerated)

Comment (nab-os) [reply src/Gui/Language/FreeCAD.ts]: Understood, it should be good now

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8048">pipeline status</a> for feature branch [PR_8048](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8048). Pipeline [725915507](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/725915507) was triggered at [d7b8402](https://github.com/FreeCAD/FreeCAD/pull/8048/commits/d7b8402cb842cb3f2b636260aa10252458e60109). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by donovaly** [comment]
Many thanks for your PR. I tested it but need a brief description what the difference of this style to the Trackball style is.

**Comment by nab-os** [comment]
For example,
Here, I started from the same face each time, and I rotated the object by moving my mouse diagonally for the same distance for both.

With the Trackball orbit style we can see that the Z axis is not aligned with the screen's Y axis:


And with the FreeTurntable orbit style, it still is:


That's because the Trackball takes the direct transformation between the last and current mouse position whereas for the FreeTurntable, the transformation is divided into two movements: around the object and up/down movement both being relative to the screen's referential.

File: src/Gui/DlgSettingsNavigation.ui [hunk 2]
```diff
@@ -7,7 +7,7 @@
     <x>0</x>
     <y>0</y>
     <width>500</width>
-    <height>391</height>
+    <height>472</height>
```
Comment (donovaly) [inline src/Gui/DlgSettingsNavigation.ui:10]: 80 pixels more for one additional setting seems strange

File: src/Gui/NavigationStyle.cpp [hunk 3]
```diff
@@ -114,33 +115,47 @@ class FCSphereSheetProjector : public SbSphereSheetProjector {
...
-                angle = -angle;
-            rot.setValue(xaxis, angle);
-        }
-        else {
-            SbVec3f zaxis(0,0,1);
-            this->worldToScreen.multDirMatrix(zaxis, zaxis);
-            if (zaxis[1] < 0) {
-                if (dif[0] < 0)
+        else if (orbit == Turntable) {
+            // 0000333: Turntable camera rotation
```
Comment (donovaly) [inline src/Gui/NavigationStyle.cpp:119]: What does the 0000333 mean? I know this is not your code but maybe you have an idea?

File: src/Mod/Tux/NavigationIndicatorGui.py [hunk 4]
```diff
@@ -514,8 +522,10 @@ def onOrbitShow():
     gOrbit.blockSignals(True)
     if pView.GetInt("OrbitStyle", 1):
         aTrackball.setChecked(True)
-    else:
+    elif pView.GetInt("OrbitStyle", 0):
         aTurntable.setChecked(True)
+    else:
```
Comment (donovaly) [inline src/Mod/Tux/NavigationIndicatorGui.py:527]: I see an issue here because when using the menu, I don't get the orbit style I want (I cannot set Turntable). Why not set an if, elif, elif to catch all possible 3 orbit styles?

**Review by donovaly — CHANGES_REQUESTED** [review]
Looks good, I only encounter an issue when setting the orbit style via the footer in the FreeCAD main Window

File: src/Gui/NavigationStyle.cpp (hunk 3 above)
Comment (nab-os) [reply src/Gui/NavigationStyle.cpp:119]: I don't know, it was there before so I didn't want to touch it.

File: src/Gui/DlgSettingsNavigation.ui (hunk 2 above)
Comment (nab-os) [reply src/Gui/DlgSettingsNavigation.ui:10]: Qt Designer modified this.
That's the first time I use it, so maybe I touched something I shouldn't have.
Should I revert this directly by editing the .ui file ?

File: src/Gui/NavigationStyle.cpp (hunk 3 above)
Comment (luzpaz) [reply src/Gui/NavigationStyle.cpp:119]: That's the old.ticket numbering format for the mantis bug tracker. It's linking to ticket: https://tracker.freecadweb.org/view.php?id=333

Comment (nab-os) [reply src/Gui/NavigationStyle.cpp:119]: Oh great, so now that we have both possibilities, should we remove this comment ?

File: src/Mod/Tux/NavigationIndicatorGui.py (hunk 4 above)
Comment (nab-os) [reply src/Mod/Tux/NavigationIndicatorGui.py]: Like this ?

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8048">pipeline status</a> for feature branch [PR_8048](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8048). Pipeline [726614332](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/726614332) was triggered at [2121a98](https://github.com/FreeCAD/FreeCAD/pull/8048/commits/2121a98bcbe502079137f8ad44137a0aeecbc6fe). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8048">pipeline status</a> for feature branch [PR_8048](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8048). Pipeline [726621831](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/726621831) was triggered at [df20075](https://github.com/FreeCAD/FreeCAD/pull/8048/commits/df200756e8afece8abcfabe0e5941b61b708daba). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by donovaly** [comment]
Many thanks. I merged and fixed the remaining issue that you changed the default orbit style: 83a7c57b

**Comment by donovaly** [comment]
There was another issue, I fixed now: 8b6797e1

**Comment by donovaly** [comment]
OK, I added it now to the release notes:
https://wiki.freecadweb.org/Release_notes_1.0#Further_user_interface_improvements

**Comment by donovaly** [comment]
There was a further bug that the currently active orbit style was not correctly displayed in the status bar. I could fix this: #8072
